### ISSUE TITLE
Missing Franca dependencies on 1.4.0

### ISSUE THREAD

**Issue by andreisanta09** [bmwcarit/joynr#28]
Hi,

I am trying to create some docker images from my Joynr based applications but when running mvn install in the docker container the Franca dependencies are not retrieved. I first noticed this yesterday and waited to see if this is a temporary issue.

_Failed to execute goal io.joynr.tools.generator:joynr-generator-maven-plugin:1.4.0:generate (generate-java) on project provider_api: Execution generate-java of goal io.joynr.tools.generator:joynr-generator-maven-plugin:1.4.0:generate failed: Plugin io.joynr.tools.generator:joynr-generator-maven-plugin:1.4.0 or one of its dependencies could not be resolved: The following artifacts could not be resolved: org.franca:org.franca.core.dsl:jar:0.10.0, org.franca:org.franca.deploymodel.dsl:jar:0.10.0, org.franca:org.franca.core:jar:0.10.0: Could not find artifact org.franca:org.franca.core.dsl:jar:0.10.0 in central (https://repo.maven.apache.org/maven2)_

There seems to be no Franca folder on the Maven Central repository (not sure if there was one before).
If I rename the Franca folder in my local .m2 repository a new one is created and populated but there are no .jar or .pom files for any of the artifacts .

![franca](https://user-images.githubusercontent.com/10113737/44719865-553ed980-aacd-11e8-89c0-d92ab289f4be.png)

I know this is a Franca artifacts issue but could be that Joynr framework needs a reference update.

For Joynr 1.5.0 the joynr backend services are also missing from the repository : 

_[ERROR] Failed to execute goal on project radio-app: Could not resolve dependencies for project io.joynr.examples:radio-app:jar:1.5.0: The following artifacts could not be resolved: io.joynr.java.messaging.bounceproxy:single-bounceproxy:war:1.5.0, io.joynr.java.backend-services:discovery-directory-jee:war:1.5.0, io.joynr.java.backend-services:domain-access-controller-jee:war:1.5.0: Failure to find io.joynr.java.messaging.bounceproxy:single-bounceproxy:war:1.5.0 in https://repo.maven.apache.org/maven2_

If I comment out these dependencies I get the same Franca missing artifacts problem as described above (with Franca version 0.13.0).

Thanks,
Andrei

**Comment by manuel-schiller** [comment]
As of joynr 1.5.0, we decided not to publish the joynr backend services anymore on maven central.
If you need them, you need to checkout joynr and run `mvn clean install` to build them locally.

About Franca dependencies:
Yes, unfortunately those are not available via maven. We therefore include them in the joynr repo (see also https://github.com/bmwcarit/joynr/blob/d89905cc1ebe004701cdb6dfed64bc05f4e535d1/wiki/franca.md#franca-idl-overview):

https://github.com/bmwcarit/joynr/tree/master/tools/generator/dependency-libs/libs/org/franca/0.13.0

You wrote you ran "mvn install in the docker container". What exactly did you do? running `mvn install` in the joynr root will both build joynr and install the Franca dependencies.

**Comment by andreisanta09** [comment]
Hi,

My bad - I must have overseen the Franca dependencies details as I have build the joynr some time ago and then I must have gotten the Franca resources.

I created a docker file inside my application which should run `mvn install` (using a maven image as base) for my solution which does not reside in the joynr sub-folders. I have added in the pom file of my solution only the joynr dependencies I needed which can also be retrieved from Maven repository (except for Franca). 

Thanks,
Andrei
